colossus/src/main/scala/colossus/protocols/memcache/Memcache.scala

```diff
@@ -10,39 +10,149 @@ import java.util.zip.{Deflater, Inflater}
+  }
+  case class Add(key: ByteString,  value: ByteString, ttl: Integer) extends MemcacheCommand {
+  
+    def bytes(compressor: Compressor) = {
+      val data = compressor.compress(value)
+      FormatCommand(ByteString("add"), key, Some(value), Some(ByteString(s"${ttl}")), Some(ByteString(s"0")))
+    }
+  }
+  object Replace {
+    def apply(key: String, value: String, ttl: Integer = 0): Replace = Replace(ByteString(key), ByteString(value), ttl)
+  }
+  case class Replace(key: ByteString, value: ByteString, ttl: Integer) extends MemcacheCommand {
+    def bytes(compressor: Compressor) = {
+      val data = compressor.compress(value)
+      FormatCommand(ByteString("replace"), key, Some(data), Some(ByteString(s"${ttl}")), Some(ByteString(s"0")))
+    }
+  }
+  
+  // Append does not take <flags> or <expiretime> but we have to provide them according to the protocol
+  object Append {
+    def apply(key: String, value: String, ttl: Integer = 0): Append = Append(ByteString(key), ByteString(value), ttl)
+  }
+
+  case class Append(key: ByteString, value: ByteString, ttl: Integer) extends MemcacheCommand {
+    def bytes(compressor: Compressor) = {
+      val data = compressor.compress(value)
+      FormatCommand(ByteString("append"), key, Some(data), Some(ByteString(s"${ttl}")), Some(ByteString(s"0")))
+    }
+  }
+
+  // Prepend does not take <flags> or <expiretime> but we have to provide them according to the protocol
+  object Prepend {
+    def apply(key: String, value: String, ttl: Integer = 0): Prepend = Prepend(ByteString(key), ByteString(value), ttl)
+  }
+  case class Prepend(key: ByteString, value: ByteString, ttl: Integer) extends MemcacheCommand {
+    def bytes(compressor: Compressor) = {
+      val data = compressor.compress(value)
+      FormatCommand(ByteString("prepend"), key, Some(data), Some(ByteString(s"${ttl}")), Some(ByteString(s"0")))
+    }
+  }
+  
+  object Cas {
+    def apply(key: String, value: String, ttl: Integer = 0): Cas = Cas(ByteString(key), ByteString(value), ttl)
+  }
+  case class Cas(key: ByteString, value: ByteString, ttl: Integer) extends MemcacheCommand {
+    def bytes(compressor: Compressor) = {
+      val data = compressor.compress(value)
+      FormatCommand(ByteString("cas"), key, Some(data), Some(ByteString(s"${ttl}")), Some(ByteString(s"0")))
+    }
   }
   case class Delete(key: ByteString) extends MemcacheCommand {
-    def bytes(c: Compressor) = ByteString("delete ") ++ key ++ RN
+    def bytes(c: Compressor) = FormatCommand(ByteString("delete"), key)
   }
   object Delete {
     def apply(key: String): Delete = Delete(ByteString(key))
   }
+  object Incr {
+    def apply(key: String, value: Integer): Incr = Incr(ByteString(key), ByteString(s"${value}"))
+  }
+  case class Incr(key: ByteString, value: ByteString) extends MemcacheCommand {
+    def bytes(c: Compressor) = ByteString("incr ") ++ key.filter(_ > ' ').slice(0,250) ++ ByteString(" ") ++ value ++ RN
```

Comment:
So Incr/Decr actually have a slightly different format that everything that uses FormatCommand, I can work on figuring out a way for them to go through the same interface though. Basically, Incr/Decr don't have an extra \r\n before the data you want to increment by.